src/python/WMCore/WMSpec/WMWorkload.py

```diff
@@ -123,6 +124,42 @@ def setStepProperties(self, assignArgs):
 
         return
 
+    def updateStepParentageMap(self):
+        """
+        _updateStepParentageMap
+        Used to update the step parentage mapping of StepChain requests at the
+        end of the assignment process, given that we might have new output
+        dataset names
+        :return: just updates the workload property: stepParentageMapping
+        """
+        topLevelTask = next(self.taskIterator())
+
+        parentMap = self.getStepParentageMapping()
+        listOfStepNames = parentMap.keys()
+        for stepName in listOfStepNames:
+            if parentMap[stepName]['OutputDatasetMap']:
+                # then there is output dataset, let's update it
+                cmsRunNumber = parentMap[stepName]['StepCmsRun']
+                stepHelper = topLevelTask.getStepHelper(cmsRunNumber)
+                for outputModuleName in stepHelper.listOutputModules():
+                    outputModule = stepHelper.getOutputModule(outputModuleName)
+                    outputDataset = "/%s/%s/%s" % (outputModule.primaryDataset,
+                                                   outputModule.processedDataset,
+                                                   outputModule.dataTier)
+
+                    # now find and replace the old dataset by the new dataset name
+                    oldOutputDset = parentMap[stepName]['OutputDatasetMap'][outputModuleName][0]
```

Comment:
Apparently an output module in a task/step has only one output dataset. Just wondering whether we should make it a string instead of a list? @ticoann

Reply:
So it does creates only one output modules per step which means parent dataset and child dataset is always one to one (at least within the same Step/TaskChain workflow.)

Reply:
Actually each step can have several output modules, but everything leads me to think that each PSet/Step output module has only one output dataset. AND, the same output module is not present twice in the same PSet/Step. @vlimant maybe you can confirm it?

Reply:
Yep, I'll make these changes on the other PR. Thanks!